### ISSUE TITLE
Granular Status Codes

### ISSUE THREAD

**Issue by boxhock** [smartcontractkit/external-adapters-js#1454]
Instead of just throwing an error with 400 or 500, we should have separate status codes for different known errors.

For example:
- If something times out we should know if it was the adapter didn't process it in time or if the request hung in flight to the data provider.
- If a data provider returns `null` data that we can't handle we should have a status code specifically for this scenario

**Comment by justinkaseman** [comment]
Pasting from DMs with André for visibility:

So If a data provider returns null data that we can't handle we should have a status code specifically for this scenario.
You could put that in https://github.com/smartcontractkit/external-adapters-js/blob/develop/packages/core/bootstrap/src/lib/external-adapter/requester.ts#L100 .
Line 113 would throw an error with a status code that we knew specifically meant that null data was returned.

If something times out, we should know if it was the adapter didn't process it in time or if the request hung in flight to the data provider. That would be in https://github.com/smartcontractkit/external-adapters-js/blob/develop/packages/core/bootstrap/src/lib/external-adapter/requester.ts#L56 .
You would want to check this error to see if it was an Axios timeout, if so return the error code designated for timeouts.

Basically for most `throw`s in bootstrap we would want to return a certain error code for it.
Feel free to choose which error code is assigned to which error.
